packages/edit-site/src/components/global-styles/shadows-edit-panel.js

```diff
@@ -301,18 +301,20 @@ function ShadowEditor( { shadow, onChange } ) {
 	const shadowParts = useMemo( () => getShadowParts( shadow ), [ shadow ] );
 
 	const onChangeShadowPart = ( index, part ) => {
-		shadowParts[ index ] = part;
-		onChange( shadowParts.join( ', ' ) );
+		const newShadowParts = shadowParts.map( ( p, i ) =>
+			i === index ? part : p
+		);
+		onChange( newShadowParts.join( ', ' ) );
 	};
 
 	const onAddShadowPart = () => {
-		shadowParts.push( defaultShadow );
-		onChange( shadowParts.join( ', ' ) );
+		const newShadowParts = [ ...shadowParts, defaultShadow ];
+		onChange( newShadowParts.join( ', ' ) );
 	};
 
 	const onRemoveShadowPart = ( index ) => {
-		shadowParts.splice( index, 1 );
-		onChange( shadowParts.join( ', ' ) );
+		const newShadowParts = shadowParts.filter( ( p, i ) => i !== index );
+		onChange( newShadowParts.join( ', ' ) );
```

Comment:
Same as above, we should be able to do it without an extra constant because there's no mutation going on here:

```suggestion
		onChange( shadowParts.filter( ( p, i ) => i !== index ).join( ', ' ) );
```